src/components/InfiniteGrids/InfiniteVideoGrid.tsx

```diff
@@ -109,11 +109,12 @@ const InfiniteVideoGrid: React.FC<InfiniteVideoGridProps> = ({
     createdAtGte,
   ])
 
+  const serializedVariables = JSON.stringify(queryVariables)
   const placeholderItems = Array.from({ length: placeholdersCount }, () => ({ id: undefined }))
   const gridContent = (
     <>
       {[...displayedItems, ...placeholderItems]?.map((video, idx) => (
-        <VideoPreview id={video.id} key={idx} showChannel={showChannel} />
+        <VideoPreview id={video.id} key={idx + serializedVariables} showChannel={showChannel} />
```

Comment:
This fixes issue #365 and it allows the videoPreviews to show its animations and new data properly when the queryVariables change; i.e. category change

Reply:
I feel like this is fixing the symptom instead of the problem. This also results in a different behaviour. What #365 describes is that on `master`, the video preview could transition between two videos without a problem. Right now it cannot, so by this change in the video grid we're basically forcing it to go into a loading state in between. I think the problem here is that currently the video preview cannot transition between two videos. If it could, like it does on `master`, my guess is that we also wouldn't need the `loadOnMount` since the previews could do a transition without load in between. What are your thoughts?

Reply:
I don't think it's a huge deal, what this does is actually makes it behave a little more like it was before as the `key` on master equals the `video.id` and we went for `idx` to make it looks better with the placeholders and adding the queries is more an in-between solution.

To the why the videoPreview cant switch from one video to the other without loadOnMount it's because we replaced the css animations with CSSSwitchTransitions which also have a key and they only trigger when we first pass a `loading=true` I haven't seen a way around that yet other than to make it load on the first time it's rendered and then set is to false on the useEffect like the loadOnMount thing